### ISSUE TITLE
Possible bug with CIPW norm ; question regarding use in interpreter

### ISSUE THREAD

**Issue by vinigpcruz** [GeoPyTool/GeoPyTool#13]
### Bug report

**Bug summary**

GeoPython, running in its stand-alone version (most recent one as far as I can tell) crashes when I tried to do a CIPW norm using the sample 'Geochemistry' .xlsx file. After the crash the terminal screen reads : 

' Traceback (most recent call last):
  File "C:\Users\vinig\OneDrive\Geologia\Softwares\GeoPyTool-Windows-2019-5-6\GeoPyTool-Windows-2019-5-6\GeoPyTool\lib\site-packages\geopytool\__init__.py", line 1373, in CIPW
    self.cipwpop.CIPW()
  File "C:\Users\vinig\OneDrive\Geologia\Softwares\GeoPyTool-Windows-2019-5-6\GeoPyTool-Windows-2019-5-6\GeoPyTool\lib\site-packages\geopytool\CIPW.py", line 303, in CIPW
    self.WholeResult = pd.concat([self.FinalResultMole,self.FinalResultWeight,self.FinalResultVolume,self.FinalResultCalced], axis=1,sort=False )
TypeError: concat() got an unexpected keyword argument 'sort'

C:\Users\vinig\OneDrive\Geologia\Softwares\GeoPyTool-Windows-2019-5-6\GeoPyTool-Windows-2019-5-6>pause
Pressione qualquer tecla para continuar. . .'

The last sentence is portuguese for : 'press any key to continue'. Of the other functions I have tried this is the only one not working.

**GeoPyTool version**

  * Windows 10 64
  * GeoPyTool version:  Most recent one I believe
  * Which function causes the error: CIPW norm
  * Other Information:  Using the sample geochemistry data file.

* Instalation was done using the 'stand-alone' version.
 

# Question

This is not a bug per se but rather a curiosity that I have seen it has been asked before. Is it possible to use GeoPython as a module (e.g., pandas) and call the functions from within a python3 interpreter? In the article you do mention that possibility and the website does feature an 'expert' installation guide. What I am wondering is how can I call the functions, for example a 'do_TAS' function of something like that. I am quite new to python (just getting started reading Think Python) so this may be a somewhat confusing question.

Last, but not least, thanks for the work.

Best Regards,

**Comment by cycleuser** [comment]
Thanks for the report on CIPW.
I just test the latest version on my local laptop but it didn't reappear. So the problem you mentioned might be caused by pandas? If so, updating pandas to latest version might be a good solution.

<img width="1147" alt="image" src="https://user-images.githubusercontent.com/6130092/106865552-4df07100-6706-11eb-8bc5-3715668b4909.png">

Originally, it was always my intention to wrap these functions into modules for people to call under the command line interpreter. But that may not be possible for years to come.

After graduating, I started working a busy schedule with very limited free time to spend with my children and provide for my family.
Other geologists in my nation did not care much for or support this and other similar open source software for geological purposes. Then this project has not been possible to get any funding to constantly continue development for long time.
It's sad that things are so unpredictable. 
The academic world is always disappointing and always lame everywhere.

After all, thank you for your encouragement and support, and wish you all the best and success in your research and studies.

**Comment by vinigpcruz** [comment]
Thanks for the the reply CycleUser. I might try updating pandas and see if it works. I really want to get around using this tool rather then the GCDkit most people use in my university. First because I know much more of python than I do of R and second because I really like your iniciative. Sad to hear that using it as a module might not be possible for years. I would love to call these functions from a module and working with dataframes in parallel from the interpreter.

Now, on to a more 'off-topic' subject; Please don't take time off your family, I think you've already done a great job by starting the project. As an early carrer research from a 'developing' country that does not care much for science I can totally relate to your comment. I started academia thinking about question, theories and the laws of the universe. The more I dig myself in the academia the more I see ego and mediocre people in positions of high autority. Don't even get me started on the very anti-ethic behaviour of journal publishers and some revierws.  Lets hope for a better academic environment in the future (even though my hopes are low).

Please stay safe and healthy.

Best Regards,

Vinícius
